2023-09-06 08:00:27,663 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-06 08:00:33,911 [INFO] shopapp.views: Rendering shop index
2023-09-06 08:10:48,894 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-06 08:10:52,303 [INFO] shopapp.views: Rendering shop index
2023-09-06 11:19:19,417 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-06 11:19:30,040 [WARNING] django.request: Not Found: /
2023-09-06 11:19:38,824 [WARNING] django.request: Not Found: /
2023-09-06 11:35:50,195 [INFO] django.utils.autoreload: /home/ilmi/PycharmProjects/python_django/mysite/shopapp/views.py changed, reloading.
2023-09-06 11:35:51,162 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-06 11:35:55,016 [ERROR] django.request: Internal Server Error: /en/shop/products/export/
Traceback (most recent call last):
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/mysite/shopapp/views.py", line 327, in get
    name = elem['naem']
           ~~~~^^^^^^^^
KeyError: 'naem'
2023-09-06 11:53:22,689 [INFO] django.utils.autoreload: /home/ilmi/PycharmProjects/python_django/mysite/mysite/settings.py changed, reloading.
2023-09-06 11:53:23,700 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-06 11:53:49,326 [ERROR] django.request: Internal Server Error: /en/shop/products/export/
Traceback (most recent call last):
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/sentry_sdk/integrations/django/views.py", line 84, in sentry_wrapped_callback
    return callback(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/mysite/shopapp/views.py", line 327, in get
    name = elem['naem']
           ~~~~^^^^^^^^
KeyError: 'naem'
2023-09-06 11:56:36,510 [INFO] django.utils.autoreload: /home/ilmi/PycharmProjects/python_django/mysite/shopapp/views.py changed, reloading.
2023-09-06 11:56:37,953 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-07 14:35:43,570 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-07 14:35:51,987 [WARNING] django.request: Not Found: /
2023-09-07 14:35:53,141 [WARNING] django.request: Not Found: /favicon.ico
2023-09-07 14:36:01,539 [INFO] shopapp.views: Rendering shop index
2023-09-07 14:49:18,466 [INFO] django.utils.autoreload: /home/ilmi/PycharmProjects/python_django/mysite/shopapp/forms.py changed, reloading.
2023-09-07 14:49:19,568 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-07 16:17:26,323 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-07 16:17:35,129 [WARNING] django.request: Not Found: /
2023-09-07 16:18:19,361 [ERROR] django.request: Internal Server Error: /admin/shopapp/product/
Traceback (most recent call last):
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/core/handlers/base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/sentry_sdk/integrations/django/views.py", line 38, in sentry_patched_render
    return old_render(self)
           ^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/sentry_sdk/integrations/django/templates.py", line 75, in rendered_content
    return real_rendered_content.fget(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/test/utils.py", line 112, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/test/utils.py", line 112, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/test/utils.py", line 112, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/test/utils.py", line 112, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/template/defaulttags.py", line 471, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/urls/base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/urls/resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'import_products_csv' not found. 'import_products_csv' is not a valid view function or pattern name.
2023-09-07 16:19:54,267 [INFO] django.utils.autoreload: /home/ilmi/PycharmProjects/python_django/mysite/shopapp/admin.py changed, reloading.
2023-09-07 16:19:55,702 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-07 16:20:30,536 [WARNING] django.request: Bad Request: /admin/shopapp/product/import-products-csv/
2023-09-07 16:23:42,845 [ERROR] django.request: Internal Server Error: /admin/shopapp/product/import-products-csv/
Traceback (most recent call last):
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 328, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.IntegrityError: NOT NULL constraint failed: shopapp_product.created_by_id

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/sentry_sdk/integrations/django/views.py", line 84, in sentry_wrapped_callback
    return callback(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/mysite/shopapp/admin.py", line 96, in import_csv
    Product.objects.bulk_create(products)
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/models/query.py", line 803, in bulk_create
    returned_columns = self._batched_insert(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/models/query.py", line 1831, in _batched_insert
    self._insert(
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/models/query.py", line 1805, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1822, in execute_sql
    cursor.execute(sql, params)
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/debug_toolbar/panels/sql/tracking.py", line 252, in execute
    return self._record(super().execute, sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/debug_toolbar/panels/sql/tracking.py", line 177, in _record
    return method(sql, params)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/sentry_sdk/integrations/django/__init__.py", line 616, in execute
    return real_execute(self, sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ilmi/PycharmProjects/python_django/venv/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 328, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.IntegrityError: NOT NULL constraint failed: shopapp_product.created_by_id
2023-09-07 16:52:26,174 [INFO] django.utils.autoreload: /home/ilmi/PycharmProjects/python_django/mysite/shopapp/views.py changed, reloading.
2023-09-07 16:52:29,756 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-07 17:02:11,544 [INFO] django.utils.autoreload: /home/ilmi/PycharmProjects/python_django/mysite/shopapp/admin.py changed, reloading.
2023-09-07 17:02:12,742 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-07 17:16:01,051 [INFO] django.utils.autoreload: /home/ilmi/PycharmProjects/python_django/mysite/shopapp/admin.py changed, reloading.
2023-09-07 17:16:03,104 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-07 17:16:24,081 [INFO] django.utils.autoreload: /home/ilmi/PycharmProjects/python_django/mysite/shopapp/views.py changed, reloading.
2023-09-07 17:16:25,135 [INFO] django.utils.autoreload: Watching for file changes with StatReloader
2023-09-07 17:16:31,727 [WARNING] django.request: Method Not Allowed: /shop/api/products/upload_csv/
2023-09-07 17:19:06,685 [WARNING] django.request: Method Not Allowed: /shop/api/products/upload_csv/
